third_party/move/move-vm/types/src/value_serde.rs

```diff
@@ -172,6 +171,50 @@ pub fn serialized_size_allowing_delayed_values(
     })
 }
 
+/// If given type has a constant serialized size (irrespective of the instance), it returns the serialized
+/// size in bytes any value would have.
+/// Otherwise it returns None.
+pub fn constant_serialized_size(ty_layout: &MoveTypeLayout) -> PartialVMResult<Option<usize>> {
+    let bcs_size_result = match ty_layout {
+        MoveTypeLayout::Bool => bcs::serialized_size(&false).map(|size| Some(size)),
+        MoveTypeLayout::U8 => bcs::serialized_size(&0u8).map(|size| Some(size)),
+        MoveTypeLayout::U64 => bcs::serialized_size(&0u64).map(|size| Some(size)),
+        MoveTypeLayout::U128 => bcs::serialized_size(&0u128).map(|size| Some(size)),
+        MoveTypeLayout::Address => bcs::serialized_size(&AccountAddress::ZERO).map(|size| Some(size)),
+        // vectors have no constant size
+        MoveTypeLayout::Vector(_) => Ok(None),
+        // enums have no constant size
+        MoveTypeLayout::Struct(MoveStructLayout::RuntimeVariants(_) | MoveStructLayout::WithVariants(_)) => Ok(None),
+        MoveTypeLayout::Struct(MoveStructLayout::Runtime(fields)) => {
+            Ok(fields
+                .iter()
+                .map(|field| constant_serialized_size(field))
+                .collect::<Result<Option<Vec<_>>, _>>()
+                .map(|o| o.map(|v| v.iter().sum()))?)
+        },
+        MoveTypeLayout::Struct(MoveStructLayout::WithFields(fields))
+        | MoveTypeLayout::Struct(MoveStructLayout::WithTypes{fields, ..}) => {
+            Ok(fields
+                .iter()
+                .map(|field| constant_serialized_size(&field.layout))
+                .collect::<Result<Option<Vec<_>>, _>>()
```

Comment:
Can this potentially allocate too much? Shall we bother to write a loop or try_for_each()?

Reply:
this is nice, but I agree - safer to not worry about it :)

Reply:
updated